File: gr/sample-and-hold-liquidities.hs
-- sample and hold liquidities (jmcc) #4

import Sound.SC3 {- hsc3 -}

shl :: UGen
shl =
    let r = mouseX KR 1 200 Exponential 0.1
        t = recip r
        c = impulse KR r 0 * 0.4
        cf = mouseY KR 100 8000 Exponential 0.1
        f = latch (whiteNoise 'α' KR * cf * 0.5 + cf) c
        p = latch (whiteNoise 'β' KR) c
        i = pan2 (sinOsc AR f 0 * decay2 c (t * 0.1) (t * 0.9)) p 1
    in combN i 0.3 0.3 2

main :: IO ()
main = audition (out 0 shl)

File: gr/sample-and-hold-liquidities.fs
\ sample and hold liquidities (jmcc) #4

: r 1 200 1 0.1 MouseX.kr ;
: t r Recip ;
: c r 0 Impulse.kr 0.4 * ;
: cf 100 8000 1 0.1 MouseY.kr ;
: f WhiteNoise.kr cf * 0.5 * cf + c Latch ;
: p WhiteNoise.kr c Latch ;
f 0 SinOsc.ar c t 0.1 * t 0.9 * Decay2 * p 1 Pan2 0.3 0.3 2 CombN play
